lib/RawPreview.php

```diff
@@ -118,21 +122,30 @@ protected function getResizedPreview($tmpPath, $maxX, $maxY)
         //tmp
         $previewImageTmpPath = dirname($tmpPath) . '/' . md5($tmpPath . uniqid()) . '.jpg';
 
-        //extract preview image using exiftool to file
-        shell_exec($this->converter . " -b -" . $previewTag . " " . escapeshellarg($tmpPath) . ' > ' . escapeshellarg($previewImageTmpPath));
-        if (filesize($previewImageTmpPath) < 100) {
-            throw new \Exception('Unable to extract valid preview data');
+        if ($previewTag === 'SourceTIFF') {
+        	// load the original file as fallback when TIFF has no preview embedded
+	        $previewImageTmpPath = $tmpPath;
+        } else {
+	        //extract preview image using exiftool to file
+	        shell_exec($this->converter . " -b -" . $previewTag . " " . escapeshellarg($tmpPath) . ' > ' . escapeshellarg($previewImageTmpPath));
+	        if (filesize($previewImageTmpPath) < 100) {
+		        unlink( $previewImageTmpPath );
+		        throw new \Exception('Unable to extract valid preview data');
+	        }
+
+	        //update previewImageTmpPath with orientation data
+	        shell_exec($this->converter . ' -TagsFromFile ' . escapeshellarg($tmpPath) . ' -orientation -overwrite_original ' . escapeshellarg($previewImageTmpPath));
         }
-        //update previewImageTmpPath with orientation data
-        shell_exec($this->converter . ' -TagsFromFile ' . escapeshellarg($tmpPath) . ' -orientation -overwrite_original ' . escapeshellarg($previewImageTmpPath));
 
         $im = Image::make($previewImageTmpPath);
         $im->orientate();
         $im->resize($maxX, $maxY, function ($constraint) {
             $constraint->aspectRatio();
             $constraint->upsize();
         });
-        unlink($previewImageTmpPath);
+	    if ($previewTag !== 'SourceTIFF') {
+		    unlink( $previewImageTmpPath );
+	    }
```

Comment:
Fix indention here and this can be merged :)

Reply:
sorry, my IDE did too much "smart" stuff. should be consistent now.